File: C#/Trivia/Trivia/GameRunner.cs
using System;

namespace Trivia
{
    public class GameRunner
    {
        private bool _notAWinner;
        private readonly IInput _input;
        private readonly IOutput _output;
        
        public GameRunner(IInput input, IOutput output)
        {
            _input = input;
            _output = output;
        }

        public static void Main(string[] args)
        {
            new GameRunner(new DefaultInput(), new DefaultOutput()).Run();
        }

        public void Run()
        {
            var aGame = new Game(_output);

            aGame.Add("Chet");
            aGame.Add("Pat");
            aGame.Add("Sue");

            do
            {
                aGame.Roll(_input.Die());

                if (!_input.ResponseIsCorrect())
                {
                    _notAWinner = aGame.WrongAnswer();
                }
                else
                {
                    _notAWinner = aGame.WasCorrectlyAnswered();
                }
            } while (_notAWinner);
        }
    }
}

File: scala/src/main/scala/com/adaptionsoft/games/trivia/runner/GameRunner.scala
package com.adaptionsoft.games.trivia.runner

import com.adaptionsoft.games.uglytrivia.Game
import java.util.Random


object GameRunner:
  var notAWinner = true

  @main def main(): Unit =
    var aGame = new Game();
    aGame.add("Chet")
    aGame.add("Pat")
    aGame.add("Sue")

    var rand: Random = new Random

    while notAWinner do
      aGame.roll(rand.nextInt(5) + 1)
      if (rand.nextInt(9) == 7) {
        notAWinner = aGame.wrongAnswer
      }
      else {
        notAWinner = aGame.wasCorrectlyAnswered
      }
